server_addon/houdini/client/ayon_houdini/plugins/create/create_hda.py

```diff
@@ -68,21 +94,100 @@ def create_instance_node(
         else:
             hda_node = to_hda
 
-        hda_node.setName(node_name)
+        # If user tries to create the same HDA instance more than
+        # once, then all of them will have the same product name and
+        # point to the same hda_file_name. But, their node names will
+        # be incremented.
+        hda_node.setName(node_name, unique_name=True)
         self.customize_node_look(hda_node)
+
+        # Set Custom settings.
+        hda_def = hda_node.type().definition()
+
+        if pre_create_data.get("use_ayon_icon"):
+            hda_def.setIcon(get_ayon_icon_filepath())
```

Comment:
Does this mean the icon path is hardcoded into the HDA? Or does this embed the icon as PNG data into the HDA definition?

Reply:
It means that if some user wanted to use the ayon icon, they can toggle that option instead of searching for the ayon icon location and add it manually in the hda type properties.

Reply:
I understood that - I was just asking whether the HDA would store it as `/path/to/ayon_icon.png` or whether it would be an embedded PNG data? Because I believe an HDA can do both.

Reply:
As far as I know, it takes a copy and embed it. 
I tested it and I found it's not embedded 😅 
what do you think ? 
![image](https://github.com/ynput/ayon-core/assets/20871534/68aacca0-61fc-4ba5-9c52-3eb199876cef)

Reply:
😵 I restarted my Houdini session and then it changed on its own.. but only the one that was published from production bundle? 
![image](https://github.com/ynput/ayon-core/assets/20871534/46ee8f27-cdf0-4519-a282-90a9cc64a7f0)

Reply:
No idea how it works in practice - at least the checkbox to the right is marked.

@fabiaserra do you know when it exactly "embeds" and whether it needs any special sauce to force do that?

Reply:
not sure when it does... I would expect that as long as you set it and then save it should be embedded if that toggle is on

Reply:
then, I'll just remove it. 
it was not requested anyways. I made because I was excited I knew new things about HDA creation and thought it was cool thing to have.